src/coreclr/src/jit/simdcodegenxarch.cpp

```diff
@@ -36,16 +36,29 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #define INSERTPS_TARGET_SELECT(i) ((i) << 4)
 #define INSERTPS_ZERO(i) (1 << (i))
 
+// ROUNDPS/PD:
+// - Bit 0 through 1 - Rounding mode
+//   * 0b00 - Round to nearest (even)
+//   * 0b01 - Round toward Neg. Infinity
+//   * 0b10 - Round toward Pos. Infinity
+//   * 0b11 - Round toward zero (Truncate)
+// - Bit 2 - Source of rounding control, 0b0 for immediate.
+// - Bit 3 - Precision exception, 0b1 to ignore. (We don't raise FP exceptions)
+#define ROUNDPS_TO_NEAREST_IMM 0b1000;
+#define ROUNDPS_TOWARD_NEGATIVE_INFINITY_IMM 0b1001;
+#define ROUNDPS_TOWARD_POSITIVE_INFINITY_IMM 0b1010;
+#define ROUNDPS_TOWARD_ZERO_IMM 0b1011;
```

Comment:
I guess now [these](https://github.com/dotnet/runtime/blob/a1bc3cc248bcd5b1edbf26cd8e4303b9e8baecdb/src/coreclr/src/jit/codegenxarch.cpp#L7276-L7286) magic numbers can be replaced with these named constants

Reply:
Where do you think those macros should go to, in order for those constants to be shared across `simdcodegenxarch.cpp` and `codegenxarch.cpp`? As someone that hadn't done a lot of C++ I'm not sure if it can go to `codegen.h` or some place.

Reply:
> 
> 
> I think probably instrsxarch.h would be a good place - other thoughts @dotnet/jit-contrib ?
> In any event, I think it's something that can be deferred if you'd prefer (and perhaps open an issue)

I would put the constants into emitxarch.h or codegen.h - they are supposed to be used together with a emitter or codegen function.

In my opinion, instrsxarch.h and other instrs*.h should be used for defining instructions' encodings only.

Reply:
@echesakovMSFT - I generally agree that instrs*.h should only be used for instruction encodings, but these are constants that are specific to the encodings. I don't think they belong in codegen.h, but emitxarch.h makes a lot of sense.

Reply:
I think I'll defer this for now. Not only that I don't fully understand the codebase yet, I cannot tell if including another header (if we were to move it to emitxarch) just for that constant is a good idea as I lack experience with C++.